pkg/provider/hpa.go

```diff
@@ -163,7 +165,7 @@ func (p *HPAProvider) updateHPAs() error {
 				}
 
 				collector, err := p.collectorFactory.NewCollector(&hpa, config, interval)
-				if err != nil {
+				if err != nil && !p.disregardIncompatibleHPAs {
```

Comment:
If you get an error here and have `disregardIncompatibleHPAs==true` then we would try to add the `nil` collector below.

Better would be to determine from the error if it was an error setting up the collector which should not happen or if the error was simply because no collector is supported for the HPA.

Reply:
I have refactored this to address your concerns. What do you think if the latest commit?